test/test_examples.sh

```diff
@@ -5,54 +5,70 @@
 
 set -e
 
-WORKSPACE=$1
+SOURCE_DIR=$1
 BUILD_DIR=$2
-INSTALL_DIR=$3
+NEW_BUILD_DIR=$3
+INSTALL_DIR=$4
 
 echo "Running: $0 $*"
 
 function print_usage() {
 	echo "$(basename $0) - test all examples standalone"
-	echo "Usage: $(basename $0) <workspace_dir> <build_dir> <install_dir> <list-of-examples-to-run>"
+	echo "Usage: $(basename $0) <source_dir> <build_dir> <new_build_dir> <install_dir> <list-of-examples-to-run>"
 }
 
-if [ "$3" == "" ]; then
+if [ "$4" == "" ]; then
 	print_usage
 	echo -e "Error: too few arguments\n"
 	exit 1
 fi
 
-if [ "$4" == "" ]; then
+if [ "$5" == "" ]; then
 	print_usage
 	echo "No examples to run!"
 	exit 0
 fi
 
-if [ ! -f $WORKSPACE/README.md ]; then
-	echo -e "error: incorrect <workspace_dir>: $WORKSPACE\n"
+if [ ! -f $SOURCE_DIR/README.md ]; then
+	echo -e "error: incorrect <source_dir>: $SOURCE_DIR\n"
 	print_usage
 	exit 1
 fi
 
-WORKSPACE=$(realpath $WORKSPACE)
+[ ! -d $NEW_BUILD_DIR ] && mkdir -p $NEW_BUILD_DIR
+[ ! -d $INSTALL_DIR ] && mkdir -p $INSTALL_DIR
+
+# Copy CMakeCache.txt and CMakeFiles from $BUILD_DIR to $NEW_BUILD_DIR
+# and update the build directory in $NEW_BUILD_DIR/CMakeCache.txt.
+cp -r -f $BUILD_DIR/CMakeCache.txt $BUILD_DIR/CMakeFiles $NEW_BUILD_DIR/
+SED_BUILD_DIR=$(echo $BUILD_DIR | sed "s/\//\\\\\//g")
+SED_NEW_BUILD_DIR=$(echo $NEW_BUILD_DIR | sed "s/\//\\\\\//g")
+sed -i "s/$SED_BUILD_DIR/$SED_NEW_BUILD_DIR/g" $NEW_BUILD_DIR/CMakeCache.txt
+
+SOURCE_DIR=$(realpath $SOURCE_DIR)
 BUILD_DIR=$(realpath $BUILD_DIR)
+NEW_BUILD_DIR=$(realpath $NEW_BUILD_DIR)
 INSTALL_DIR=$(realpath $INSTALL_DIR)
 
-shift 3
+echo "SOURCE_DIR=$SOURCE_DIR"
+echo "BUILD_DIR=$BUILD_DIR"
+echo "NEW_BUILD_DIR=$NEW_BUILD_DIR"
+echo "INSTALL_DIR=$INSTALL_DIR"
+
+shift 4
 EXAMPLES="$*"
 echo "Examples to run: $EXAMPLES"
 echo
 
-cd ${BUILD_DIR}
-echo "DIR=$(pwd)"
-
 set -x
+cd $NEW_BUILD_DIR
+cmake $SOURCE_DIR -DCMAKE_INSTALL_PREFIX="$INSTALL_DIR"
 make -j$(nproc) install
```

Comment:
i talk about ```make DESTDIR=/package/stage install```

maybe this link will be better:
https://cmake.org/cmake/help/latest/envvar/DESTDIR.html#envvar:DESTDIR

Reply:
The command:
`$ make DESTDIR=/package/stage install`
installs all files to the `${DESTDIR}/${CMAKE_INSTALL_PREFIX}` directory in fact, so it would not work.
You would have to run `cmake` and change `CMAKE_INSTALL_PREFIX` anyway.

Reply:
Actually it can work ... I will try ...

Reply:
It seems it works :-) Thanks!
Done.